src/components/Export.tsx

```diff
@@ -196,138 +290,288 @@ export default function Export(inputProps: ExportButtonProps): ReactElement {
         setTimeout(() => stopRecording(true), 750);
       },
-      onRecordedFrameCallback: (frame: number) => {
+      onRecordedFrame: (frame: number) => {
         // Update the progress bar as frames are recorded.
         setPercentComplete(Math.floor(((frame - min) / (max - min)) * 100));
       },
-    });
+      onError: handleError,
+    };
+
+    // Initialize different recorders based on the provided options.
+    switch (recordingMode) {
+      case RecordingMode.VIDEO_MP4:
+        recorder.current = new Mp4VideoRecorder(props.setFrame, props.getCanvas, recordingOptions);
+        break;
+      case RecordingMode.IMAGE_SEQUENCE:
+      default:
+        recorder.current = new ImageSequenceRecorder(props.setFrame, props.getCanvas, recordingOptions);
+        break;
+    }
+    recorder.current.start();
   };
 
-  const numExportedFrames = Math.max(Math.ceil((customMax - customMin + 1) / frameIncrement), 1);
+  //////////////// RENDERING ////////////////
+
+  const tooltipTrigger: ("hover" | "focus")[] = ["hover", "focus"];
+
+  const videoQualityOptions = [
+    { label: "High", value: VideoBitrate.HIGH },
+    { label: "Med", value: VideoBitrate.MEDIUM },
+    { label: "Low", value: VideoBitrate.LOW },
+  ];
+
+  const isWebCodecsEnabled = Mp4VideoRecorder.isSupported();
+  const customRangeFrames = Math.max(Math.ceil((customMax - customMin + 1) / frameIncrement), 1);
+
+  const totalFrames = rangeMode === RangeMode.CUSTOM ? customRangeFrames : props.totalFrames;
+  const totalSeconds = totalFrames / fps;
+
+  // Gets the total duration as a MM min, SS sec label.
+  // Also adds decimal places for small durations.
+  const getDurationLabel = (): string => {
+    const durationMin = Math.floor(totalSeconds / 60);
+    const durationSec = totalSeconds - durationMin * 60;
+
+    let timestamp = "";
+    if (durationMin > 0) {
+      timestamp += durationMin.toString() + " min, ";
+    }
+    // Format seconds to hundredths if less than 10 seconds
+    if (durationMin === 0 && durationSec < 10) {
+      // Round digits to 2 decimal places
+      const roundedSeconds = Math.round(durationSec * 100) / 100;
+      timestamp += roundedSeconds.toFixed(2) + " sec";
+    } else {
+      timestamp += Math.floor(durationSec).toString() + " sec";
+    }
+    return timestamp;
+  };
+
+  const getApproximateVideoFilesizeMb = (): string => {
+    // From experimentation, filesize is dependent on duration and bitrate.
+    // It scales linearly with the bitrate until a maximum filesize is hit,
+    // which seems to depend on the video dimensions.
+
+    // Video quality is bitrate in bits/second
+    const maxVideoBitsDuration = totalSeconds * videoBitsPerSecond;
+
+    // Experimentally-determined compression ratio (bits per pixel). This may not apply for all
+    // datasets, depending on image complexity.
+    // For high bitrates, the video filesize will be constrained by duration.
+    const compressionRatioBitsPerPixel = 2.77;
```

Comment:
shouldn't this be relative to number and spacing of keyframes and also the bit rate (videoBitsPerSecond)?   Also, just wondering how accurate the final estimate is in practice.

Reply:
Updated this comment block to illustrate why this is needed and increased the value to overshoot estimates.